server/database/couchdbSeeder.js

```diff
@@ -0,0 +1,204 @@
+      };
+      rooms.push(roomObj);
+
+    }
+
+    const ownerInfo = {
+      ownerId: ownerIdVal,
+      name: faker.name.firstName(),
+      joinedDate: '2020-04-23',//faker.date.past(),
+      reviewsCount: faker.random.number(),
+      isIdentityVerified: faker.random.boolean(),
+      isSuperHost: faker.random.boolean(),
+      responseRate: faker.random.number({ min: 93, max: 100 }),
+      responseTime: faker.random.arrayElement(responseType),
+      profilePic: faker.image.imageUrl(),
+      duringStay: faker.lorem.sentences(),
+      hostDescription: faker.lorem.sentences(),
+      language: faker.random.arrayElements(languages, faker.random.number({ min: 0, max: 3 })),
+      roomDetails: rooms
+    }
+
+    create(ownerInfo, err => {
+      if (err) {
+        throw err;
+      } else {
+        console.log('Owner Inserted', i);
+        var d = new Date();
+        let localtime = d.toLocaleTimeString('en-US', { hour12: false });
+        console.log('Start  localtime:', localstarttime)
+        console.log('End  localtime:', localtime)
+      }
+    })
+
+  }
+
+
+}
+
+const create = (owner, cb) => {
+  owners.insert(owner, owner.ownerId, cb);
+
+}
+
+const bulkUploadToDB = async (count) => {
+
+  const batchSize = 10000;
+  let currentCount = 0;
+  let currentRoomCount = 1;
+  let bulkArrayData = [];
+  var d = new Date();
+  let localstarttime = d.toLocaleTimeString('en-US', { hour12: false });
+
+  while (count > 0) {
+    console.log('count : ', count);
+
+    for (var i = 0; i < batchSize; i++) {
+      currentCount++;
+      //let ownerIdVal = faker.random.uuid();
+      //generate rooms per user
+      let roomsPerUser = faker.random.number({
+        'min': 1,
+        'max': 1
+      });
+      let rooms = [];
+      for (var j = 0; j < roomsPerUser; j++) {
+        let roomObj = {
+          roomId: currentRoomCount++,
+          ownerId: currentCount,
+          roomName: faker.lorem.word()
+        };
+        rooms.push(roomObj);
+      }
+
+      const ownerInfo = {
+        ownerId: currentCount,
+        name: faker.name.firstName().trim(),
+        joinedDate: faker.date.past().toISOString(),
+        reviewsCount: faker.random.number(),
+        isIdentityVerified: faker.random.boolean(),
+        isSuperHost: faker.random.boolean(),
+        responseRate: faker.random.number({ min: 93, max: 100 }),
+        responseTime: faker.random.arrayElement(responseType),
+        profilePic: faker.image.imageUrl(),//faker.random.arrayElement(awsUrls),
+        duringStay: faker.lorem.sentences(),
+        hostDescription: faker.lorem.sentences(),
+        language: faker.random.arrayElements(languages, faker.random.number({ min: 0, max: 3 })),
+        roomDetails: rooms
+      }
+      bulkArrayData.push(ownerInfo);
+      count--;
+
+    }
+    await owners.bulk({ docs: bulkArrayData });
+    bulkArrayData = [];
+
+  }
+
+  return localstarttime;
+
+}
+//load 10 mil records
+generateAWSUrls()
```

Comment:
Looks good. Did you run into any challenges promise chaining the generation of AWS URLs with database insertions? I approached the two separately, so I was just curious.

Reply:
When I generated the urls on the fly  for each owner, it was taking long time. Then I added check to do it for every 10,000 but didn't help much. So, I just generated the 1000 image urls  and picked it in random for data load. Generating 1000 urls is taking around 10-12 min for me. Still need to look into way to optimize.